specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,551 @@
+        Double Value;
+
+        // Gets/Sets the Value for the Progress String. Binds to {progressValueString} in progress xml tag
+        String ValueStringOverride;
+
+        // Gets/Sets the Value for the Status. Binds to {progressStatus} in progress xml tag
+        String Status;
+    }
+
+    // The Notification User Setting or Notification Group Policy Setting
+    enum AppNotificationSetting
+    {
+        Enabled, // Notification is not blocked by settings or group policy
+        DisabledForApplication, // Notification is blocked by a user defined App Setting
+        DisabledForUser, // Notification is blocked by a user defined Global Setting
+        DisabledByGroupPolicy, // Notification is blocked by Group Policy
+        DisabledByManifest, // Notification is blocked by a setting in the manifest. Only for packaged applications.
+    };
+
+    // The Result for a Notification Progress related operation
+    enum AppNotificationProgressResult
+    {
+        Succeeded, // The progress operation succeeded
+        AppNotificationNotFound, // The progress operation failed to find a Notification to process updates
+    };
+
+    // The Priority of the Notification UI associated with it's popup in the Action Centre
+    enum AppNotificationPriority
+    {
+        Default, // The notification should have default behavior in terms of delivery and display priority during connected standby mode.
+        High, // The notification should be treated as high priority. For desktop PCs, this means during connected standby mode the incoming notification can turn on the screen for Surface-like devices if it doesn't have a closed lid detected.
+    };
+
+    runtimeclass AppNotification
+    {
+        AppNotification(Windows.Data.Xml.Dom.XmlDocument payload);
+
+        // Unique identifier used to replace a notification within a group.
+        String Tag;
+
+        // Unique identifier for a Notification group in the app
+        String Group;
+
+        // A unique identifier for the Notification generated by the platform.
+        UInt32 Id { get; };
+
+        // The notification Xml Payload
+        Windows.Data.Xml.Dom.XmlDocument Payload{ get; };
+
+        // Gets or sets additional information about the Notification progress.
+        AppNotificationProgressData Progress;
+
+        // Gets or sets the time after which a Notification should not be displayed.
+        Windows.Foundation.DateTime Expiration;
+
+        // Indicates whether the Notification will remain in the Action Center after a reboot.
+        Boolean ExpiresOnReboot;
+
+        // Gets or sets the priority for a Notification.
+        // Hints on how and at what urgency level a notification should be presented to the user (whether to wake up the screen, etc).
+        AppNotificationPriority Priority;
+
+        // Gets or sets whether a Notification's pop-up UI is displayed on the user's screen.
+        Boolean SuppressDisplay;
```

Comment:
Sounds like it's suppressing the physical display (esp. since the `Priority` talks about turning on the display). I assume it's a "negative" property name so that the default (false) means "show the thing" (double-negative) but you can have a positive name and set the "default" value to true in the constructor. For example, `ShowNotification` or `DisplayNotification` etc. 

More broadly, I'm not sure why you'd send a toast and not show it. Is this so it shows up in Action Center but doesn't annoy me first with the popup? If so, then maybe something like `AddDirectlyToNotificationCenter` or something (not the best name...)

Reply:
The old platform name for this was `SurpressPopup`. Given we seem to talk about these as "popups", was there a reason `SuppressDisplay` was chosen instead? Or `SuppressBanner` since the Settings app calls these "notification banners"? I agree that `Display` is an odd choice.

Reply:
@jonwis Any thoughts on this point? This feedback was from your prior recommendation. I'm totally ok with rolling back :)

Reply:
No comments on this. So keeping as is.